jdaviz/tests/test_subsets.py

```diff
@@ -60,12 +60,14 @@ def test_region_from_subset_3d(jdaviz_app):
 
 
 def test_region_from_subset_profile(jdaviz_app, spectral_cube_wcs):
-    data = Data(flux=np.ones((256, 128, 128)), label='Test 1D Flux', coords=spectral_cube_wcs)
+    data = Data(flux=np.ones((128, 128, 256)), label='Test 1D Flux', coords=spectral_cube_wcs)
     jdaviz_app.data_collection.append(data)
 
     jdaviz_app.add_data_to_viewer('spectrum-viewer', 'Test 1D Flux')
 
-    jdaviz_app.get_viewer("spectrum-viewer").apply_roi(XRangeROI(5, 15.5))
+    viewer = jdaviz_app.get_viewer('spectrum-viewer')
+    viewer.state.x_att = jdaviz_app.data_collection['Test 1D Flux'].id['Frequency']
```

Comment:
What does it mean that you need to add this line to the test from a dev perspective?

Reply:
I actually might not need this now that I moved the relevant logic from the parser to the helper, let me check...

Reply:
Ah, right. The answer is that changing the x-axis of the `spectrum-viewer` automatically gets triggered via `AddDataMessage` in the Cubeviz helper class, but these tests aren't using the Cubeviz helper class, they're initializing things at a lower level via jdaviz.Application. I think the best thing to do here is to update to test using the helper class, to more accurately represent actual use cases.